addons/survey/static/src/js/survey_form.js

```diff
@@ -725,11 +725,14 @@ publicWidget.registry.SurveyFormWidget = publicWidget.Widget.extend(SurveyPreloa
                 case 'matrix':
                     if (questionRequired) {
                         var subQuestionsIds = $questionWrapper.find('table').data('subQuestions');
+                        const missingRowIds = [];
                         subQuestionsIds.forEach(function (id) {
                             if (!((questionId + '_' + id) in data)) {
                                 errors[questionId] = constrErrorMsg;
+                                missingRowIds.push(id)
```

Comment:
Hello @amdi-odoo,
I think you can simplify the code even further :smile: If we don't use the `_highlightRows` function elsewhere, we can maybe directly patch the dom at each iteration without accumulating the missing row ids ? We will then have a tiny diff and a simpler code. As a bonus point, you will also make a reviewer happy somewhere in the world :blush:

Note: If you need to have access to the context of the widget to call `this.el`, you can transform the anonymous function of the `forEach` loop to an anonymous arrow function (i.e: `() => {}`).